pkg/operators/uidgidresolver/uidgidresolver.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package uidgidresolver provides an operator that enriches events by looking
+// up uid and gid resolving them to the corresponding username and groupname.
+package uidgidresolver
```

Comment:
Document the limitations here too: https://github.com/inspektor-gadget/inspektor-gadget/pull/2751#issuecomment-2071838165